vm/os-windows.cpp

```diff
@@ -304,6 +304,54 @@ void handle_ctrl_c() {
   SetConsoleCtrlHandler(factor::ctrl_handler, TRUE);
 }
 
+static volatile bool stop_on_ctrl_break = false;
+static HANDLE ctrl_break_thread = NULL;
+
+static DWORD WINAPI ctrl_break_thread_proc(LPVOID mainThread) {
+  bool ctrl_break_handled = false;
+  while (true) {
+    if (factor::stop_on_ctrl_break) {
+      if (GetAsyncKeyState(VK_CANCEL) >= 0) { /* Ctrl-Break is released. */
+        ctrl_break_handled = false;  /* Wait for the next press. */
+      } else if (!ctrl_break_handled) {
+        /* This proc runs in its own thread without stopping the main
+           thread. Since in practice nobody uses the multi-VM stuff yet,
+           we just interrupt the first VM we can get. */
+        FACTOR_ASSERT(thread_vms.size() > 0);
+        THREADHANDLE thd = thread_vms.begin()->first;
+        factor_vm* vm = thread_vms.begin()->second;
+        /* Check if the VM thread has the same Id as the thread Id of the
+           currently active window. Note that thread Id is not a handle. */
+        DWORD fgThreadId = GetWindowThreadProcessId(GetForegroundWindow(),
+                                                    NULL);
+        if (!atomic::load(&vm->safepoint.fep_p)
+            && (fgThreadId == vm->thread_id)
+        ) {
+          atomic::store(&vm->skip_debugger_p, true);
+          vm->safepoint.enqueue_fep(vm);
+          wake_up_thread(thd); /* Try to wake up the thread. */
+          ctrl_break_handled = true;
+        }
+      }
+    }
+    Sleep(100);
+  }
```

Comment:
Now I've experimented a lot and 100 ms makes to many ctrl+break presses get lost on my machine. If it doesn't affect performance 30 ms or less would be a better polling interval.